### ISSUE TITLE
Is it only supported in.vue files? Can CSS methods be used in.ts files

### ISSUE THREAD

**Issue by SGAMERyu** [Tahul/pinceau#5]
when i use css method in single ts file 
Vite reported an error compiling the library

```
 'extname' is not exported by __vite-browser-external, imported by ../../node_modules/.pnpm/unplugin@0.9.5_6k4qr4f4opqyvhlnzjpej7fvtq/node_modules/unplugin/dist/index.mjs
piece-ui:dev: file: C:/Users/qixiyu/Desktop/code/piece-ui/node_modules/.pnpm/unplugin@0.9.5_6k4qr4f4opqyvhlnzjpej7fvtq/node_modules/unplugin/dist/index.mjs:8:9
piece-ui:dev:  6: 
piece-ui:dev:  7: // src/esbuild/utils.ts
piece-ui:dev:  8: import { extname } from "path";
piece-ui:dev:              ^
piece-ui:dev:  9:
piece-ui:dev: 10: // node_modules/.pnpm/@jridgewell+sourcemap-codec@1.4.11/node_modules/@jridgewell/sourcemap-codec/dist/sourcemap-codec.mjs
```

**Comment by Tahul** [comment]
I could totally support that.

Thanks for reporting it.

Could you provide me an example of how you tried to use it in a `.ts` component, considering there is no `<style>` there?

**Comment by SGAMERyu** [comment]
> 我完全可以支持。
> 
> 感谢您报告它。
> 
> `.ts`考虑到那里没有，您能否提供一个示例，说明您如何尝试在组件中使用它`<style>`？

I'm trying to build a component library, so I looked at the code organization of popular component libraries in the Vue community, such as ElementPlus and Vant, which extract style files into a single file for Demand Import
The code directory is as follows

```
button
     -  button.style.ts
     - button.vue
```

Here is the link to my sample repository https://github.com/SGAMERyu/pinceau-vue

**Comment by SGAMERyu** [comment]
> I could totally support that.
> 
> Thanks for reporting it.
> 
> Could you provide me an example of how you tried to use it in a `.ts` component, considering there is no `<style>` there?


I'm trying to build a component library, so I looked at the code organization of popular component libraries in the Vue community, such as ElementPlus and Vant, which extract style files into a single file for Demand Import
The code directory is as follows

```
button
     - button.style.ts
     - button.vue
```

Here is the link to my sample repository https://github.com/SGAMERyu/pinceau-vue

**Comment by SGAMERyu** [comment]
I think it might be better to Use directly on the component template,
The merge approach may result in users using different Cssdeclarations, which are not very useful for developers to experience with an orthogonal language like css. When a style problem occurs, developers need to read and debug in different Cssdeclarations

**Comment by SGAMERyu** [comment]
@Tahul 
Of course, I am very excited about the latest version of Pinceau as well. Perhaps we can create a dedicated discussion area to engage in discussions.

**Comment by SGAMERyu** [comment]
I have a question about the above method, that is, how to handle the current vue component style and import style objects merge

> @grbd @SGAMERyu ;
> 
> This is something that will get supported very soon after the release of #103
> 
> I would love you both to discuss a potential DX for that feature in that thread.
> 
> I am thinking of:
> 
> ```ts
> // my-styling-file.ts
> export const myStyle = css({
>    ...myCss
> })
> 
> // MyVueComponent.vue
> 
> <style lang="ts">
> import { myStyle } from 'my-styling-file.ts'
> 
> css({
>    ...myStyle
> })
> </style>
> ```
> 
> I would love to explore potential ways of using these.
> 
> Let me know if you're still interested!

**Comment by Tahul** [comment]
What do you mean by handling current Vue component style?

Another solution would be to merge options that way:

```
const myCss = css({
   div: { backgroundColor: 'red' }
}, [myStyle])

console.log(myCss) // '.my-css-class .my-style-class'
```

**Comment by SGAMERyu** [comment]
@Tahul 
There may be style conflicts between the styles of the current vue component and the imported css styles
example
```
// my-styling-file.ts
export const myStyle = css({
   div: { background: 'red' }
})

// MyVueComponent.vue

<style lang="ts">
import { myStyle } from 'my-styling-file.ts'

css({
  div: { background: red }
   ...myStyle
})
</style>
```
This can cause the developer to find style conflicts in multiple files

**Comment by Tahul** [comment]
@SGAMERyu ; I think using this feature would merge both declarations in a single CSS object, then stringify it.

In your exemple, the background would be red.

That is why I suggest the 2nd syntax over object spreading, it is also easier to handle at compiler-level.

**Comment by SGAMERyu** [comment]
@Tahul Yes, the second approach is better, and clearly tells the developer that the current style is merged

**Comment by SGAMERyu** [comment]
@Tahul：I have another small question that I would like to ask about. The formatting function of prettier in vscode does not seem to format pinceau on an prettier
![image](https://github.com/Tahul/pinceau/assets/17926105/56a78bf8-b59c-40e2-9fc1-a6f4dc77b65e)
Indentation is handled correctly when formatting the ts file
![image](https://github.com/Tahul/pinceau/assets/17926105/63159b53-467c-4131-b1f8-28a2a8ae17f6)
Does prettier not handle the behavior of functions on the style tag, or does my own Settings not format them

If prettier is not supported, I would like to support prettier's formatting for pinceau
